manul/src/combinators/chain.rs

```diff
@@ -359,11 +375,11 @@ where
                     .map(|round_id| round_id.group_under(1))
                     .collect::<BTreeSet<_>>();
 
-                // If there are no next rounds, this is the result round.
-                // This means that in the chain the next round will be the entry round of the second protocol.
-                if next_rounds.is_empty() {
-                    next_rounds.insert(C::EntryPoint2::entry_round().group_under(2));
+                if round.as_ref().may_produce_result() {
```

Comment:
```suggestion
                if round.as_ref().produces_result() {
```

I think `produces_result` is a better name.

Reply:
It won't be accurate though: a round may either produce a result, or transition to another (error) round. We are only interested if there's a possibility that this round may terminate the protocol, so that we know what messages to expect next in a chained protocol situation.